### DISCUSSION TITLE
Remove Current Timestamp from energyplus --version

### DISCUSSION THREAD

**Discussion by Myoldmopar** [NREL/EnergyPlus#9494] [Polls]
We don't need to print the current date and time when a user just wants the version.  It's silly.  Should we remove it?

**Comment by mjwitte** [comment]
Users need the date and time to confirm if outputs have been updated after inputs.

**Reply by Myoldmopar** [reply]
I'm not saying the current date and time shouldn't be used in timestamps in actual output.  The scope of this is just when you specifically call EnergyPlus with the `--version` flag.  Which only reports the current time, which doesn't give you any more information than if you just check the current time.

**Reply by mjwitte** [reply]
Well, that wasn't clear (probably because I never use that). Then nevermind.

**Reply by Myoldmopar** [reply]
Sorry, I think if GitHub had formatted the title as code `energyplus --version` it would have been more obvious, but it just left the backticks in there and looked funny, so I removed them.

**Reply by rraustad** [reply]
![image](https://user-images.githubusercontent.com/5778156/174084316-e38c2f17-2fb2-45bf-88d9-4911f2fdfcb1.png)

**Reply by Myoldmopar** [reply]
But I need to know _when_ you executed that call to request the version.  

JK, yes, that's a more reasonable form from OS.  I'm not sure if they are parsing the `--version` output, or if it's baked into the build, but either way, that's more like it.  Kinda surprised it doesn't include the SHA.

**Comment by JasonGlazer** [comment]
Maybe it should be expanded to show the version of third party modules that have separate versions like eigen, SQLite, zlib, re2, etc..

**Reply by Myoldmopar** [reply]
That's valid, I spot checked a few command line tools and found that several of them print additional information like copyright and authorship:

```
$ cut --version
cut (GNU coreutils) 8.30
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Written by David M. Ihnat, David MacKenzie, and Jim Meyering.

$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Right now the `--help` argument already prints out the version of Python that E+ is linked to:

```
$ ep221 --help
EnergyPlus, Version 22.1.0-ed759b17ee, YMD=2022.06.17 09:56
PythonLinkage: Linked to Python Version: "3.8.12 (default, Oct 18 2021, 14:07:50) [GCC 9.3.0]"
Built on Platform: Ubuntu20.04_x86_64
Usage: energyplus [options] [input-file]
Options:
  -a, --annual                 Force annual simulation
  -c, --convert                Output IDF->epJSON or epJSON->IDF, dependent on
...
```

I'm open to printing a nice structured output to print all the versions of things:

```
$ ep221 --version
EnergyPlus: Version 22.1.0-ed759b17ee
BuiltOnPlatform: Ubuntu20.04_x86_64
PythonLinkage: 3.8.12 (default, Oct 18 2021, 14:07:50) [GCC 9.3.0]
SQLite: Something
Kiva: Something
Foo: Bar
```

That would be a little bigger job than just cleaning out the current date/time.  Other tools also do this a little more succintly:

```
$ geany --version
geany 1.36 (built on 2020-03-22 with GTK 3.24.14, GLib 2.64.1)
```

And again, some things don't print the dependency versions at all in the regular `--version` call:

```
$ man --version
man 2.9.1
```

My current proposition would be to take out the current date/time from the version/help outputs, because that is simply not needed.  At all.  Then separately, add a new flag like `--all-versions` that will print out the full list of dependency versions.  What do you think of that approach?

**Reply by JasonGlazer** [reply]
Sounds like a great approach.
